controller/handler/badge_test.go

```diff
@@ -86,8 +86,13 @@ func TestBadge(t *testing.T) {
 			ExpectedCode: http.StatusOK,
 		},
 		{
-			Name:         "badge-uptime-with-invalid-duration",
-			Path:         "/api/v1/endpoints/core_backend/uptimes/3d/badge.svg",
+			Name:         "badge-uptime-with-invalid-duration-zero-length",
+			Path:         "/api/v1/endpoints/core_backend/uptimes/0d/badge.svg",
+			ExpectedCode: http.StatusBadRequest,
+		},
+		{
+			Name:         "badge-uptime-with-invalid-duration-too-short",
+			Path:         "/api/v1/endpoints/core_backend/uptimes/d/badge.svg",
 			ExpectedCode: http.StatusBadRequest,
 		},
```

Comment:
Thank you very much for your reply, but I am quite busy these days and it will take me a few days to continue the development of this feature.

Some of my recent commits on this branch were just to experiment with some ideas. I realized that there are a lot of design choices and the final code has to ensure quality and performance.

Thanks again for your advice and careful code review!